docs/releases/maintenance-policy.md

```diff
@@ -1,30 +1,31 @@
 # Maintenance policy
 
-Our goal is to ensure that the latest InvenioRDM release is supported with bug and security fixes for minimum one year after the release date, and possibly longer.
-We aim at having a new release every year. We strive our best to ensure that upgrades between versions are fairly straight-forward to ensure users follow our latest releases.
+Our goal is to ensure that the latest InvenioRDM release is supported with bug and security fixes for a minimum of one year after its release date, and 6 months after the next latest release. This promotes stability and gives adopters a dependable transition period. We aim at having a new release every year. We strive our best to ensure that upgrades between versions are straight-forward to ensure users follow our latest releases.
 
-The maintenance policy is striving to strike a balance between maintaining a rock solid secure product while ensuring that users migrate to latest releases and ensuring that we have enough resources to actually support the maintenance policy.
+Our maintenance policy strives to strike a balance between supporting a rock-solid secure product, shipping new features + refactoring old code, and ensuring that we have enough resources to do all this. It pushes users to adopt the latest version, so that they can benefit from continued support. Most new features are behind configuration flags, so they can be gradually adopted though.
 
 ## Policy
 
 Starting with the v12 release, we have shifted from the Long-Term Support (LTS) and Short-Term Support (STS) model to aiming for one stable major release per year.
 
-**Major release:** Major versions such as v12 allow us to introduce new features, make backward incompatible changes and remove deprecated features in a progressive manner. Only the latest minor-level release for each major version is supported. A major version release is supported until the next major version release.
+**Major release:** Major versions, such as v12, introduce new features, make backward incompatible changes and remove deprecated features in a progressive manner. Only the latest minor-level release for each major version is supported. A major version release is supported for 6 months past the next major version release.
```

Comment:
> make backward incompatible changes...

I am wondering if this always true. It feels like that we mainly want to make a new release with new features in (hard to bring them back to existing releases), and we might not introduce backward incompatible changes.

It is more about that you will most probably have to run a migration script to upgrade dependencies and, sometimes, data model. Do we consider this backward incompatible changes?
What I mean is that you might not have to change any of your customizations.

Reply:
I think I see what you mean and I added some language to detail the implications of a major version for the end-user of invenio-app-rdm:
- it *could* bring backwards incompatible changes (aka breaking changes) but doesn't have to
- like you alluded to, the development team sees them as culmination of added features (save point of sorts) that it guarantees the maintenance of.

That's probably what's most useful to know for the InvenioRDM instance operators. Thanks!